client/state/breadcrumb/reducer.js

```diff
@@ -0,0 +1,35 @@
+import { withStorageKey } from '@automattic/state-utils';
+import {
+	BREADCRUMB_RESET_LIST,
+	BREADCRUMB_UPDATE_LIST,
+	BREADCRUMB_APPEND_ITEM,
+} from 'calypso/state/action-types';
+import { combineReducers, withSchemaValidation } from 'calypso/state/utils';
+import { breadcrumbSchema } from './schema';
+
+// Stores the complete list of breadcrumbs in an array,
+export const items = withSchemaValidation( breadcrumbSchema, ( state = [], action ) => {
+	switch ( action.type ) {
+		case BREADCRUMB_RESET_LIST:
+			return [];
+		case BREADCRUMB_UPDATE_LIST:
+			return [ ...action.items ];
+		case BREADCRUMB_APPEND_ITEM:
+			// Don't append if it is the same as the last item
+			// eslint-disable-next-line no-case-declarations
+			const currentLastItem = state.at( -1 ) || {};
+			if ( currentLastItem.label === action.item?.label ) {
+				return [ ...state ];
+			}
+
+			return [ ...state, action.item ];
+	}
+
+	return state;
+} );
+
+const combinedReducer = combineReducers( {
+	items,
+} );
+
+export default withStorageKey( 'breadcrumb', combinedReducer );
```

Comment:
Can you elaborate on the reason we want the breadcrumb in the persisted state?

Reply:
Sure, I have added an explanation to the issue itself and I'll paste it here as well:

> When the user clicks on a plugin card, there is a redirect to the Plugin Details page, which lead us to two ways of sending info:
> * via URL
> * via global state (redux) *

> In order to not pollute the URL with internal state, I would propose to save the breadcrumbs into the global state, which would allow us to solve the breadcrumbs problem in a more generic way.

source: https://github.com/Automattic/wp-calypso/issues/58501#issuecomment-1017803179

Reply:
I don't object adding the breadcrumb to redux state. I am curious if we need to also add them in the persisted state. Are we doing any "hard" redirect which would re-init the redux state and thus we need the persistence?

https://github.com/Automattic/wp-calypso/blob/af49fc7325532c45020fc629b68d5fb90d531e2d/docs/data-persistence.md#L1-L202

Reply:
aha, so this is what `withStorageKey` does 🤭

Submitted a new version without the persistence https://github.com/Automattic/wp-calypso/pull/60491/commits/2d8364df95badf4f5e794ea4c83fff549b87d20f